src/main/software/amazon/event/ruler/GenericMachine.java

```diff
@@ -56,7 +61,13 @@ public class GenericMachine<T> {
      */
     private final SubRuleContext.Generator subRuleContextGenerator = new SubRuleContext.Generator();
 
-    public GenericMachine() {}
+    public GenericMachine() {
+        this(new GenericMachineConfiguration.Builder().build());
```

Comment:
Do you see value in accepting the pattern similar to AWS SDK where rather than exposing `GenericMachineConfiguation`, we make the publicly avilable `builder()` method within this class and let folks use it to build their object. It reduces the need for folks to learn about one more class and let us change our minds for class name between `GenericMachineConfiguration`, `Configuration`, or something else.

Reply:
We can mark the `GenericMachine()` as deprecated to ensure folks are gradually moving to the golden-path. I don't have a good data around it, but this line from the benchmarks class made me wonder if we'd regret the class name in future and want to change it. 

```
Machine machine = new Machine(new GenericMachineConfiguration.Builder().withAdditionalNameStateReuse(true).build());
```

If that feels overkill, then let's keep things as is. Otherwise, we can use a builder() method for time-being.